beamline.rst

```diff
@@ -1,91 +1,57 @@
 Beamline API
 ============
 
-Functionality that incorporates several different instruments or
-that does not necessarily belong to one particular instrument.
+Functionality that are considered to belong to the beamlone as a whole,
+incorporating several different instruments or that does not necessarily belong
+to one particular instrument.
 
 
 List of Actuators:
 ------------------
 
-Below a list of all generic actuators across the UI (see general.rst),
-separated by data type.
-
-NB The actuators could be subdivided by component (beamline, diffractometer, etc.),
-but since the actuators can be accessed through the get_actuator(), get_actuators(),
-and set_actuator_value() top-level functions, there is no need to locate them
-within individual components. Component-specific access, such as
-
-detector.set_detector_distance(value)
-
-value = detector.get_detector_distance()
-
-could be added for convenience, but are basically superfluous,
-given that you can do
-
-set_actuator_value('detector_distance', value)
-
-value = get_actuator('detector_distance').value
+Below a list of actuators that, from an instrumentation perspective, are
+considered to belong to the beamline as a whole. Such actuators include, but
+are not limited to, actuators that have an effect on properties directly
+associated with the beamline. Such properties are for instance resolution,
+energy and resolution.
 
 
 **Actuators - Type float:**
 
     *Centring motors*
 
-        Omega, Kappa, Phi, AlignmentX,  AlignmentY,  AlignmentZ,
-        CentringX, CentringY, CentringZ,  Focus
-
-        # NB the capitalized case is to distinguish them from the current lower-case names
-             ('Phi' matches 'kappa_phi', not 'phi')
-
-        # NB There are too many motor names here, and maybe we should shorten the list.
-             It should be OK, though if the same motor is pointed to
-             by two different names. The problems is how the names match.
-             If you do not have 'CentringZ', which motor should you use during
-             centring?
-             Is 'Focus' always the same as 'AlignmentX'?
-             Does 'CentringZ' always match the same alignment motor
-             or does it depend on beamline geometry?
+        omega, kappa, kappa-phi, alignment-ver, alignment-hor, alignment-depth,
```

Comment:
Why use dashes '-'? That means you have to remember whether this one is dash or underscore. Unless there is a clear rule for what goes where, should we not stick to underscore separators?

Reply:
oh, just though it was easier to read, but we can go with underscore. I though that distinguishing between kappa motors with capital letters was a bit too subtle. And then I just wanted to be consistent

Reply:
These names have some problems. In Hamburg, where Omega is vertical and 'sampx', 'sampy' are rotated twenty-odd degrees away from the beam direction at omega=0, it would not map well to reality. 

- Which alignment and centring axis is parallel to omega 1) if omega is horizontal? 2) if omega is vertical?

- Which translation axes are parallel to the beam 1) if  omega is horizontal, 2) if omega is vertical? 

- How do I find out which alignment axis is identical to fous (if any) and which alignment axis is identical to one (which) of the centring axes?

In the GPhL code we only need to know which are the three centring axes, and which one is parallel to omega. If possible it would be nice to have a set of generic names that could be intuitively mapped to the variously named motors at different beamlines. If that is not possible, we can put the mapping in a config file, though.

Reply:
Well for the UI what matters is what horizontal, vertical to the microscope and the user should not need to know how the diffractometer is oriented.

Focus is always focus on the microscope, if I change focus I really don't want to move the sample in any direction even if that is what happens in some implementations.

The set of motors you are speaking about are more for experiment setup and internal control

Reply:
At least that is my view and I think that would actually further harmonize the UI, its a different matter for kappa angles though ... A set of diffractometer specific motors might be needed in a few instances though ...